### PR TITLE
Hot007 issue307 update guide

### PR DESCRIPTION
**Pull request by hot007**
Add content to Instructors Guide to Python Ecology lesson, particularly the final (SQL) section.
(part of instructor check-out).

### CODE REVIEW

**Comment by maxim-belkin** [comment]
Hi @hot007! Thank you for the PR!

I like your changes but I'm curious as to why you change some headings, e.g., "05-merging-data" to "05-merging-data-with-pandas". In this particular example, episode title is "Combining DataFrames with Pandas" and the filename is `05-merging-data.md` (which, I believe, was used for titles). I think it makes sense to use either episode titles OR names of the files. I suspect using abbreviated episode titles might be confusing.

Other than that -- it looks great! thank you!

**Comment by hot007** [comment]
Hi Maxim,
Ah, I was trying to get the title closer to the episode title (explicitly saying "with pandas" but definitely fine to reject that bit!
No worries!
-Claire

File: _extras/guide.md [hunk 1]
```diff
@@ -740,3 +750,99 @@ plt.show()
...
+  FROM surveys INNER JOIN plots ON surveys.plot_id = plots.plot_id INNER JOIN species ON \
+  surveys.species_id = species.species_id WHERE surveys.year>=1998 AND surveys.year<=2001 \
+  AND ( surveys.sex = "M" OR surveys.sex = "F")')
+
+print(len(cur.fetchall()))
+
+# Close the connection
+con.close()
+~~~
+
```
Comment (maxim-belkin) [inline _extras/guide.md:777]: There should be no empty lines between `~~~` and `{: .language-python}` lines

File: _extras/guide.md [hunk 2]
```diff
@@ -740,3 +750,99 @@ plt.show()
...
+{: .language-python}
+
+Answer: 5546 records are found.
```
Comment (maxim-belkin) [inline _extras/guide.md:780]: you can use
```
~~~
...
~~~
{: .output}
```
approach

File: _extras/guide.md [hunk 3]
```diff
@@ -740,3 +750,99 @@ plt.show()
...
+# Turn the plot_id column values into column names by pivoting
+df2 = df2.pivot(index='year',columns='plot_id')['SUM(weight)']
+df = pd.merge(df1, df2, on='year')
+
+# Verify that result of the SQL queries is stored in the combined dataframe
+print(df.head())
+
+con.close()
+~~~
+
```
Comment (maxim-belkin) [inline _extras/guide.md:806]: Please remove this line

File: _extras/guide.md [hunk 4]
```diff
@@ -740,3 +750,99 @@ plt.show()
...
+3  3596.0  ...  1071.0  529.0  5877.0  5075.0  691.0  5523.0  321.0  3763.0   
+4  4946.0  ...  1083.0  176.0  5050.0  4773.0  410.0  5379.0  600.0  5268.0   
+
+      23      24  
+0  169.0     NaN  
+1    NaN     NaN  
+2  137.0   901.0  
+3  742.0  4392.0  
+4   57.0  3987.0  
+~~~
```
Comment (maxim-belkin) [inline _extras/guide.md:832]: you can use `{: .output}` or `{: .source}`

**Comment by maxim-belkin** [comment]
Sounds good. Please update the PR

> Hi Maxim,
> Ah, I was trying to get the title closer to the episode title (explicitly saying "with pandas" but definitely fine to reject that bit!
> No worries!
> -Claire

Excellent! Could you please update the PR accordingly then? I think these headings can be episode titles, file names (without .md), or "Episode X: Episode title". Using episode titles make the most sense to me but I don't mind if you keep them as they were.

Thank you for your work and let me know if you need help or advice with anything.

Maxim

**Comment by hot007** [comment]
Okay, I've made the requested changes I hope, please let me know if I missed anything. Thanks for the feedback, I'm learning about this {: .} syntax, that's really helpful, apologies I had it wrong, the preview for me isn't parsing those bits so I couldn't see it wasn't working. 
I haven't changed the other section headings though, do we really want to call one of the sections "07-visualization-ggplot-python" when `ggplot` has been replaced with `plotnine`? Or should we rename that episode file? Similarly for Eps 6, 8 and 9, whose filenames are now substantially out of line with the episode titles.  Actually I will make another commit rolling back my changes to the headings, but just noting the filenames probably should be changed so that changes like this can be made in the instructor's guide.

**Comment by maxim-belkin** [comment]
> I haven't changed the other section headings though, do we really want to call one of the sections "07-visualization-ggplot-python" when `ggplot` has been replaced with `plotnine`? Or should we rename that episode file? Similarly for Eps 6, 8 and 9, whose filenames are now substantially out of line with the episode titles. Actually I will make another commit rolling back my changes to the headings, but just noting the filenames probably should be changed so that changes like this can be made in the instructor's guide.

My personal preference is to use "Ep. #: Episode Title" headings but what you did here makes perfect sense. Of course, we should keep episode file names in sync with the content, e.g. ggplot -> plotnine, etc., but these discrepancies should be addressed in separate PRs (1 file change per PR).

I'm going to merge this PR as it seems to be ready. Thank you for the excellent contribution! If you'd like to work on any of the open issues -- let me know and I'll be happy to help!

🏅

**Pull request merged**